### DISCUSSION TITLE
Windows Batch File to Combine Images, but Keep Same Name?

### DISCUSSION THREAD

**Discussion by Konsider** [ImageMagick/ImageMagick#4742] [General] [answered]
Hi :)

Ok, so I have a folder of images. They are all 800x800 pixels. (image1.png, image2.png, etc)

I wish to merge them, centered, onto a background image which is 900x900 pixels. (back.png)

I wish to be able to keep the same name that the images already have, so I am guessing I would need to specify a new directory. (newdirectory/image1.png, newdirectory/image2.png, etc)

Can someone please show me the batch file (Windows) I would need in order to do this?

Thank you so much! :)

**Comment by snibgo** [comment, marked as answer]
I suggest you break this down into two problems.

First, figure out the IM command that does what you want for a single file. For example:
```
magick back.png image1.png -composite outdir\image1.png
```

Then put that in a shell loop that repeats the IM command for every image. For example (BAT syntax):
```
for %%F in (*.png) do magick back.png %%F -composite outdir\%%F
```

**Reply by Konsider** [reply]
That is super helpful -  really appreciate that! :) I will try to see if I can get it working... hopefully I won't have to come back and bother you with another question ;) Cheers, mate!

**Reply by Konsider** [reply]
@snibgo The files combine together using the batch file, however, the problem is... the 800x800 image is not centered in the 900x900 background. Could I trouble you for info as to how I would get it centered? It appears to just be added to the top left of the background at the moment.

**Reply by Konsider** [reply]
If there is no way of centering the image, could we add a line to the batch file before the composite that would resize the files to 900x900 first? Of course, this would have to be done with a transparent background? Then again, a resize might not work, as it would stretch the contents... it would have to be a "canvas size"?

**Reply by fmw42** [reply]
add -gravity center before -composite

**Reply by Konsider** [reply]
It works! Thanks so much for your help! :)
